File: openmp-wg-pII-2015/omp_task.c
#include <stdio.h>
#include <omp.h>

int main(){
    #pragma omp parallel
    #pragma omp single
    {
        printf("Thread %d is creating tasks\n",omp_get_thread_num());
        #pragma omp task
            printf("Hello 1 from thread %d\n",omp_get_thread_num());
        #pragma omp task
            printf("Hello 2 from thread %d\n",omp_get_thread_num());
    }
}

File: openmp-wg-pII-2016/omp_task.f90
program omp_task
    use omp_lib
!$OMP PARALLEL
!$OMP SINGLE
        write(*,"(A,I4,A)") "Thread ",omp_get_thread_num()," is creating tasks"
!$OMP TASK
        write(*,"(A,I4)") "Hello 1 from thread",omp_get_thread_num()
!$OMP END TASK
!$OMP TASK
        write(*,"(A,I4)") "Hello 2 from thread",omp_get_thread_num()
!$OMP END TASK
!$OMP END SINGLE
!$OMP END PARALLEL

end program
